apps/web/src/routes/Messages/MessageForm/MessageForm.js

```diff
@@ -43,49 +48,48 @@ const MessageForm = React.memo(forwardRef((props, ref) => {
     props.sendIsTyping(true)
   }, STARTED_TYPING_INTERVAL)
 
-  const {
-    className,
-    currentUser,
-    messageText,
-    onFocus,
-    pending,
-    placeholder = t('Write something...')
-  } = props
+  useEffect(() => {
+    console.log('MessageForm state', {
+      messageText: props.messageText,
+      pending: props.pending,
+      hasRef: !!textareaRef.current,
+      disabled: textareaRef.current?.disabled
+    })
+  }, [props.messageText, props.pending])
 
-  if (pending) return <Loading />
   return (
     <form
-      className={cn('w-full max-w-[750px] fixed bottom-0 flex gap-3 px-2 shadow-md p-2 border-2 border-foreground/15 shadow-xlg rounded-t-xl bg-card pb-4 transition-all', className, { 'border-focus': hasFocus })}
+      className={cn('w-full max-w-[750px] fixed bottom-0 flex gap-3 px-2 shadow-md p-2 border-2 border-foreground/15 shadow-xlg rounded-t-xl bg-card pb-4 transition-all', props.className, { 'border-focus': hasFocus })}
       onSubmit={handleSubmit}
     >
-      <RoundImage url={get('avatarUrl', currentUser)} medium />
+      <RoundImage url={get('avatarUrl', props.currentUser)} medium />
       <TextareaAutosize
-        value={messageText}
+        value={props.messageText}
         className='text-foreground bg-transparent w-full my-2 focus:outline-none mt-0'
-        ref={(tag) => (textareaRef.current = tag)}
+        ref={textareaRef}
         minRows={1}
         maxRows={8}
         onChange={handleOnChange}
         onKeyDown={handleKeyDown}
-        onFocus={() => { setHasFocus(true); onFocus() }}
+        onFocus={() => { setHasFocus(true); props.onFocus?.() }}
```

Comment:
does this onFocus?.() work? I've never seen that syntax

Reply:
Made it more explicit what's happening

Reply:
I just didnt know you can add a ? after a function before calling it like that, super cool

Reply:
AI is teaching us new things